storage/innobase/log/log0log.cc

```diff
@@ -885,9 +867,9 @@ log_group_file_header_flush(
 		  ? LOG_HEADER_FORMAT_10_3
 		  : LOG_HEADER_FORMAT_10_2));
 
-	buf = *(group->file_header_bufs + nth_file);
+	// man 2 open suggests this buffer to be aligned by 512 for O_DIRECT
+	MY_ALIGNED(512) byte buf[OS_FILE_LOG_BLOCK_SIZE] = {0};
```

Comment:
For me, `man 2 open` says ‘typically 512 bytes’. I think that we should use `MY_ALIGNED(OS_FILE_LOG_BLOCK_SIZE)` here, in case we some day increase the logical redo log block size.
For what it is worth, I believe that nowadays physical sector sizes of 2048 bytes are common. But then again, they could be represented to the operating system as 512 bytes.

Reply:
> For what it is worth, I believe that nowadays physical sector sizes of 2048 bytes are common. 

I guess the best we can do is to wait for some user with fancy filesystem to fill a bug in Jira :)